src/components/App/App.jsx

```diff
@@ -12,34 +12,74 @@ import Main from "../Main/Main.jsx";
 import Footer from "../Footer/Footer.jsx";
 import ProtectedRoute from "../ProtectedRoute/ProtectedRoute.jsx";
 import LoginForm from "../LoginForm/LoginForm.jsx";
-import RegisterForm from "../RegisterForm/Registerform.jsx";
+import RegisterForm from "../RegisterForm/RegisterForm.jsx";
 import InfoPopup from "../InfoPopup/InfoPopup.jsx";
 import SavedNews from "../SavedNews/SavedNews.jsx";
 import MobileMenu from "../MobileMenu/MobileMenu.jsx";
 import { CurrentPageContext } from "../../contexts/CurrentPageContext.jsx";
+import { CurrentUserContext } from "../../contexts/CurrentUserContext";
 import { IsLoggedInContext } from "../../contexts/IsLoggedInContext.jsx";
 import { deleteItem, saveItem, getSavedItems } from "../../utils/api.js";
+import * as auth from "../../utils/auth";
 import { getNews } from "../../utils/newsApi.js";
+
 import "./App.css";
 
 function App() {
-  const [isLoggedIn, setIsLoggedIn] = useState(true);
+  const [currentUser, setCurrentUser] = useState(null);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
   const [activeModal, setActiveModal] = useState("");
   const [currentPage, setCurrentPage] = useState("home");
   const [searchResults, setSearchResults] = useState([]);
   const [isLoading, setIsLoading] = useState(false);
+  const [isLoggedInLoading, setIsLoggedInLoading] = useState(true);
   const [keyword, setKeyword] = useState("");
   const [savedItems, setSavedItems] = useState([]);
   const [searchError, setSearchError] = useState(false);
+  const [serverError, setServerError] = useState({});
+  const navigate = useNavigate();
 
   const handleLogin = ({ email, password }) => {
-    console.log(activeModal);
+    if (!email || !password) {
+      return;
+    }
+    return auth
+      .authorize({ email, password })
+      .then((res) => {
+        setIsLoggedIn(true);
+        localStorage.setItem("jwt", res.token);
+        setCurrentUser({ name: res.name, _id: res._id });
+        closeActiveModal();
+      })
+      .catch((err) => {
+        setServerError({
+          ...serverError,
+          loginError: "Incorrect email or password",
+        });
+        console.error("Failed to login", err);
+      });
   };
 
   const handleRegistration = ({ email, password, name }) => {
-    console.log(activeModal);
+    auth
+      .register({ email, password, name })
+      .then((res) => {
+        setIsLoggedIn(true);
+        localStorage.setItem("jwt", res.token);
+        setCurrentUser({ name: res.name, _id: res._id });
+        closeActiveModal();
+      })
+      .catch((err) => {
+        setServerError({
+          ...serverError,
+          regError: "A user with this email already exists",
+        });
+        console.error("Failed to register", err);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js
  function handleSubmit(request) {
    // start loading
    setIsLoading(true);
    request()
      // we need to close only in `then`
      .then(handleCloseModal) 
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
      .catch(console.error)
      // and in finally we need to stop loading
      .finally(() => setIsLoading(false));
  }

```


You need to pass the **request** into the call of `handleSubmit` like that:

```js

  // here is an example
  const handleAddItem = (item) => {
    // here we create a function that returns a promise
    const makeRequest = () => {
      // `return` lets us use a promise chain `then, catch, finally`
      return addNewItem(item).then((item) => {
        setClothingItems([item, ...clothingItems]);
      });
    };
    // here we call handleSubmit passing the request
    handleSubmit(makeRequest);
  };

```

And for the 14th sprint:

```js
 // here is an example
 function handleProfileFormSubmit(inputValues) {
  // here we create a function that returns a promise 
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return api.editProfile(inputValues).then(setCurrentUser);
  }
  // here we call handleSubmit passing the request
  handleSubmit(makeRequest);
}


```

So, this way you can remove a lot of code duplicating